lib/backend/dynamo/dynamodbbk.go

```diff
@@ -391,7 +391,7 @@ func (b *Backend) getAllRecords(ctx context.Context, startKey []byte, endKey []b
 			return nil, trace.Wrap(err)
 		}
 		result.records = append(result.records, re.records...)
-		if len(result.records) >= limit || len(re.lastEvaluatedKey) == 0 {
+		if (limit != 0 && len(result.records) >= limit) || len(re.lastEvaluatedKey) == 0 {
```

Comment:
So the issue was that we would just always enter this loop upon hitting the 1MB limit and never attempted to fetch another one? lgtm if so but please add a comment with explanation and wait for @fspmarshall's review as well.